backend/src/main/kotlin/app/ehrenamtskarte/backend/migration/migrations/V1_InitializeMigrations.kt

```diff
@@ -0,0 +1,15 @@
+package app.ehrenamtskarte.backend.migration.migrations
+
+import app.ehrenamtskarte.backend.migration.Migration
+import app.ehrenamtskarte.backend.migration.Statement
+
+internal class V1_InitializeMigrations : Migration() {
```

Comment:
In my opinion, the migration table should be created by the migration framework and not by a migration itself.
It just does not make sense design-wise, since if you change the schema of the migration table and its usage in `MigrationUtils`, these changes need to be present from the first migration on (e.g. when setting up a new system), but this first migration is supposed to be immutable. So if you define your migration schema inside a migration, you will never be able to change it, which kind of defies the point of a migration.
So I would suggest to use `SchemaUtils.create()` somewhere inside the migration utils in this specific case. Most likely you will never change the migration schema anyways, but just in case you do, you are flexible enough to handle the transition in the code.

Reply:
I think I agree :D So the changes to the migrations table should probably be handled separately.

> So I would suggest to use SchemaUtils.create() somewhere inside the migration utils in this specific case. Most likely you will never change the migration schema anyways, but just in case you do, you are flexible enough to handle the transition in the code.

I think it is reasonable to create the migrations table and do any other Migrations-Table-Migrations (:D)  just before we apply any actual migrations (but still inside the transaction, so if migration fails, we can safely go back to the old backend code (with the old migrations table))

Reply:
OK, but I would not handle migration-schema changes inside separate migrations, because then we would have to create a migrations-migration table to keep track of which migration-migrations are already applied :joy:

Reply:
Yep! I guess, changes to the migration schema will happen rarely enough to handle them in the code directly without keeping track of them in another table :D
Also, once these changes have been applied in prod, we can drop the code for migrating to new new migration schema :D